libs/topology/src/topology.cpp

```diff
@@ -1521,9 +1521,9 @@ namespace hpx { namespace threads {
         std::size_t num_of_cores_;
     };
 
-    std::size_t hardware_concurrency()
+    unsigned int hardware_concurrency()
     {
         util::static_<hw_concurrency, hardware_concurrency_tag> hwc;
-        return hwc.get().num_of_cores_;
+        return static_cast<std::size_t>(hwc.get().num_of_cores_);
```

Comment:
Is this supposed to be a cast into `std::size_t` ?

Reply:
Most likely not (even if it does no harm)! Thanks for noticing!

Reply:
See #4583 for a fix.